### PR TITLE
radix/store: update lastMeta every commit

### PR DESCRIPTION
**Pull request by pinheadmz**
I'm going down the rabbit hole on Urkel Tree state recovery (ie. a HNS reorg over a tree interval) and I noticed we have these caches for previously seen tree roots:

`getHistory()`

https://github.com/handshake-org/urkel/blob/547ab1e6213addd1168ae8b442e57e999369392a/lib/radix/store.js#L762-L765

`_readHistory()`

https://github.com/handshake-org/urkel/blob/547ab1e6213addd1168ae8b442e57e999369392a/lib/radix/store.js#L786-L789

I disabled them out of curiosity because I wanted to see Urkel to follow the Meta pointers saved to disk. However, without the cache injecting old roots totally fails.

Currently, Urkel loads the last state from disk and then assumes all new committed tree roots are saved in this cache. I don't think there's any reason to assume the cache will ever be prematurely cleared, but as a reviewer it might seem like a cache is just an optimization and not actually a requirement. The test I added covers an edge case where a developer may inadvertently clear the root cache from the Urkel store. If that happens, old roots can still be injected by following the chain of Meta pointers since this patch updates the `store.lastMeta` on every commit.

### CODE REVIEW

File: lib/radix/store.js [hunk 1]
```diff
@@ -577,6 +577,8 @@ class Store {
     await this.flush();
     await this.sync();
 
+    this.lastMeta = this.state;
+    this.lastMeta.metaPtr = state.metaPtr;
```
Comment (pinheadmz) [inline lib/radix/store.js:581]: check this, may be removed

**Review by nodech — APPROVED** [review]

**Comment by nodech** [comment]
I think this issue applies to `optimized` and `trie` as well. Do you want to update them as well ?

**Comment by pinheadmz** [comment]
> I think this issue applies to `optimized` and `trie` as well. Do you want to update them as well ?

Nope ;-) I'm going to merge this PR as is, and then we can talk about maintaining Urkel on a branch without the unused variants
